kolibri/core/assets/src/views/tab-link/index.vue

```diff
@@ -0,0 +1,109 @@
+<template>
+
+  <li>
+    <router-link :to="link" class="tab" :class="{ 'tab-has-icon-and-title': type === 'icon-and-title' }">
+
+      <div v-if="type === 'icon' || type === 'icon-and-title'" class="tab-icon">
+        <ui-icon :icon="icon" ariaLabel="title" class="icon"/>
+      </div>
+
+      <div v-if="type === 'title' || type === 'icon-and-title'" class="tab-title">
+        {{ title }}
+      </div>
+
+    </router-link>
+  </li>
+
+</template>
+
+
+<script>
+
+  const ValidateLinkObject = require('kolibri.utils.validateLinkObject');
+
+  module.exports = {
+    props: {
+      type: {
+        type: String,
+        validator(type) {
+          return ['title', 'icon', 'icon-and-title'].includes(type);
+        },
+        required: true,
+      },
```

Comment:
This looks okay, but I'm a little confused why `type` is necessary as an input prop rather than a computed prop.

Couldn't we just make `title` and `icon` optional, and then just try to "do the right thing" internally in the component?